### PR TITLE
driver verify bisect

### PR DESCRIPTION
**Pull request by richagadgil**

### CODE REVIEW

File: src/driver/verify.cpp [hunk 1]
```diff
@@ -237,6 +238,57 @@ void verify_reduced_program(const program& p,
...
+                            verify::tolerance tols)
+{
+    const auto* mm = p.get_main_module();
+    auto right         = std::distance(mm->begin(), mm->end());
+    std::size_t left = 0;
+    bool passed;
+
+    std::cout << "Bisect Verify steps: " << right << std::endl;
+    while (left <= right) {
+        std::size_t mid = left + (right - left) / 2;
```
Comment (pfultz2) [inline src/driver/verify.cpp:281]: You need to skip the instructions that are const foldable.

Comment (pfultz2) [reply src/driver/verify.cpp:281]: You can probably create a vector to do the mapping:

```cpp
std::vector<std::size_t> trim_map;
std::size_t i = 0;
for(const auto& ins:*mm)
{
    if(not ins.can_eval())
        trim_map.push_back(i);
    i++;
}
```

Then do the bisection over `trim_map` and then use it to get the actual trim value: `verify_bisect(p, trim_map[mid], t, options, vo, inputs, tols)`.

File: src/dom_info.cpp [hunk 2]
```diff
@@ -55,13 +55,25 @@ struct module_visitor
...
+        if(options.ignore_constants)
+        {
+            copy_if(v.get_children(ins), std::back_inserter(children), [&](auto child) {
+                return not child->can_eval();
+            });
+        }
+        else
+        {
+            children = v.get_children(ins);
+        }
```
Comment (pfultz2) [inline src/dom_info.cpp:76]: Let me revert these changes to the dominator since its not needed anymore.

File: src/driver/verify.cpp [hunk 3]
```diff
@@ -237,6 +241,93 @@ void verify_reduced_program(const program& p,
...
+        {
+            left = mid + 1;
+        }
+        else
+        {
+            failed = trim;
+            right  = mid - 1;
+        }
+    }
+    if(failed != -1)
```
Comment (pfultz2) [inline src/driver/verify.cpp:325]: I would do `if (failed > 0)`.

**Review by shivadbhavsar — APPROVED** [review]
